pkg/templates/git.go

```diff
@@ -214,6 +214,23 @@ func syncTemplateFromRef(
 		source += "//" + repo.SubPath
 	}
 
+	reference, err := vcs.GetRepoRef(r, ref)
+	if err != nil {
+		return err
+	}
+
+	switch {
+	case reference.Name().IsBranch():
+	case reference.Name().IsTag():
+	case reference.Name().IsRemote():
+	case reference.Name().IsNote():
+	default:
+		ref = reference.Hash().String()
+		if len(ref) > 7 {
+			ref = ref[:7]
+		}
```

Comment:
Can you explain the cause of the problem? What happen if we don't trim the hash?

Reply:
There are two consideration:

- Our version list query will try to compare version order by convert version tag to int array(compare every digits). If we keep the full hash, it will convert to integer  that out of range of integer. 
- The short hash like github will make it easier to select template version.
<img width="1373" alt="image" src="https://github.com/seal-io/walrus/assets/85595852/e97a63bd-5b02-4310-9287-a1324e094ce6">

Reply:
Updated. Modify version order sql, only sort semver version.